themes/src/main/resources/theme/keycloak.v2/login/register.ftl

```diff
@@ -46,5 +45,39 @@
             </div>
 
         </form>
+        <script type="module">
+            import { validatePassword } from "${url.resourcesPath}/js/password-policy.js";
+
+            const template = `
+            <div class="pf-v5-c-form__helper-text" aria-live="polite">
+                <div class="pf-v5-c-helper-text">
+                    <div class="pf-v5-c-helper-text__item pf-m-error">
+                        <ul class="pf-v5-c-helper-text__item-text">
+                            {errors}
+                        </ul>
+                    </div>
+                </div>
+            </div>`
+
+            const activePolicies = [
+                { length: { value: ${passwordPolicies.length!-1}, error: "${msg('invalidPasswordMinLengthMessage')}"} },
+                { maxLength: { value: ${passwordPolicies.maxLength!-1}, error: "${msg('invalidPasswordMaxLengthMessage')}"} },
+                { lowerCase: { value: ${passwordPolicies.lowerCase!-1}, error: "${msg('invalidPasswordMinLowerCaseCharsMessage')}"} },
+                { upperCase: { value: ${passwordPolicies.upperCase!-1}, error: "${msg('invalidPasswordMinUpperCaseCharsMessage')}"} },
+                { digits: { value: ${passwordPolicies.digits!-1}, error: "${msg('invalidPasswordMinDigitsMessage')}"} },
+                { specialChars: { value: ${passwordPolicies.specialChars!-1}, error: "${msg('invalidPasswordMinSpecialCharsMessage')}"} }
+            ].filter(n => Object.values(n)[0].value !== -1);
+
+            document.getElementById("password").addEventListener("change", function() {
```

Comment:
We can use fancy JS syntax here if we like :)
```suggestion
            document.getElementById("password").addEventListener("change", () => {
```

Reply:
You don't need to see https://github.com/keycloak/keycloak/pull/32846/files/ebedd42a5ee6a43c8f4cc3f186eb0ed3e908d8b5#r1756432033